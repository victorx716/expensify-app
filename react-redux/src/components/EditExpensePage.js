import React from 'react';
import {Link} from 'react-router-dom';

const EditExpensePage = (props) => {
  console.log(props)
  return (
    <div>
      edit expense with id of {props.match.params.id}
    </div>
  );
};

export default EditExpensePage;