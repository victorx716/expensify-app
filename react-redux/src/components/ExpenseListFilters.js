import React from 'react'
import { connect } from 'react-redux'
import { DateRangePicker } from 'react-dates'
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate} from '../actions/filters'
import uuid from 'uuid'

class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null,
  };

  onDatesChange = ({ startDate, endDate}) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  }
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }))
  }
  render() {
    return (
      <div>
        <input 
          type="text" 
          value = {this.props.filters.text} 
          onChange={(e) => { 
            props.dispatch(setTextFilter(e.target.value));
            }}
        />
        <select value ={this.props.filters.sortBy} onChange={(e) => {
          if (e.target.value === 'date') {
            props.dispatch(sortByDate())
          } else if (e.target.value === 'amount'){
            props.dispatch(sortByAmount())
          }
          }}>
          <option value="date" >Date</option>
          <option value = "amount">Amount</option>
        </select>
        
        <DateRangePicker
          startDateId = {uuid()}
          endDateId = {uuid()}
          startDate = {this.props.filters.startDate}
          endDate = {this.props.filters.endDate}
          onDatesChange = {this.onDatesChange}
          focusedInput = {this.state.calendarFocused}
          onFocusChange = {this.onFocusChange}
          showClearDates = {false}
          numberOfMonths = {1}
          isOutsideRange = {() => false}
        />
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
}

// hi 
export default connect(mapStateToProps)(ExpenseListFilters);