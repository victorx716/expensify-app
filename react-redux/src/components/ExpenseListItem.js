import React from 'react';
// description, amount, createdAt

const ExpenseListItem = ({ description, amount, createdAt}) => {
  <div>
    <h3>{description}</h3>
    <p>{amount} - {createdAt}</p>
  </div>
}

export default ExpenseListItem;