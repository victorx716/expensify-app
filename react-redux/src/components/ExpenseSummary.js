import React from 'react'

const ExpenseSummary = () => {
  <div>
    hi
  </div>
}


export default ExpenseSummary;