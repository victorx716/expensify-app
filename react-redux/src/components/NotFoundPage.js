import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundPage = () => (
  <div>
    404 - <Link href="/">Home</Link>
  </div>
)

export default NotFoundPage;