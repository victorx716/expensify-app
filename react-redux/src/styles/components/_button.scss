.login-button {
  background: rgba(6, 129, 167, 0.904);
  color: white;
  width: 80%;
  height: 5vh;
  border-radius: 3px;
  border: none;
  font-size: $font-size-large;
  font-weight: 300;
}