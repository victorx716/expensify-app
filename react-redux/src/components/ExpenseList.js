import React from 'react';
import { connect } from 'react-redux'; 
import ExpenseListItem from './ExpenseListItem'

const ExpenseList = (props) => (
  <div>
    <h1> Expense List</h1>
    {props.expenses.map((expense) => {
      <li>{expense.description} + {expense.amount} 
        + {expense.createdAt}
      </li>
    })}
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: state.expenses,
    filters: state.filters
  }
};

export default connect(mapStateToProps)(ExpenseList);