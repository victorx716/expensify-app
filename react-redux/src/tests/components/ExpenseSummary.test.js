import React from 'react'
import { shallow } from 'enzyme'
import {ExpenseSummary} from '../../components/ExpenseSummary'

test('should correctly render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal = {554}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with > 1 expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={3} expensesTotal = {85994}/>);
  expect(wrapper).toMatchSnapshot();
});