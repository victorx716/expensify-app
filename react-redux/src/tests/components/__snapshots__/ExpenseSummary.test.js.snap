// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with > 1 expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $859.94
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $5.54
  </h1>
</div>
`;
